specs/derivation.md

```diff
@@ -974,46 +979,171 @@ Some network upgrades require automated contract changes or deployments at speci
 To automate these, without adding persistent changes to the execution-layer,
 special transactions may be inserted as part of the derivation process.
 
-#### Eclipse: L1Block predeploy upgrade
+#### Ecotone
 
-The `L1Block` contract is upgraded to process the new Eclipse L1-data-fee parameters and L1 blob base-fee.
+The block before the Ecotone hardfork, contains the following transactions in this order:
 
-The `L1Block` is called in the very first transaction of the block,
-and parsed to retrieve the L1 block attributes.
+- L1 Attributes Transaction
+- User deposits from L1
```

Comment:
We want to ensure that we do not go over the block gas limit if users include many deposits in this block.

See the invariant here: https://github.com/ethereum-optimism/optimism/blob/e57ad63a23619e5da880c5140e33948d3e765dc5/packages/contracts-bedrock/src/L1/SystemConfig.sol#L243C61-L243C61

All of the upgrade txs must fit within `systemTxMaxGas` or we do our upgrade txs in the second block following an L1 origin update such that there are no user deposits to compete with

Reply:
The default value for `systemTxMaxGas` can be found here: https://github.com/ethereum-optimism/optimism/blob/e57ad63a23619e5da880c5140e33948d3e765dc5/op-chain-ops/genesis/layer_one.go#L29

It is 1 million but also configurable - we don't want to rely on this value being set to a particular thing, otherwise its another footgun

Reply:
We should clarify the assumptions with the gas config here that guarantee the upgrade is safe

Reply:
We can likely do all of the upgrade txns in < 2 million gas, including the EIP-4788 deploy. Once we have the contract changes merged in we can get a more accurate estimate.

cc @anikaraghu @EvanJRichard